Userland/Libraries/LibGfx/ImageFormats/JPEGLoader.h

```diff
@@ -7,13 +7,26 @@
 
 #pragma once
 
+#include <AK/Array.h>
+#include <AK/HashMap.h>
 #include <AK/MemoryStream.h>
+#include <AK/Optional.h>
 #include <LibGfx/ImageFormats/ImageDecoder.h>
+#include <LibGfx/ImageFormats/JPEGHuffmanTable.h>
 
 namespace Gfx {
 
 namespace JPEG {
 struct LoadingContext;
+
+using QuantizationTables = Array<Optional<Array<u16, 64>>, 4>;
```

Comment:
If this becomes public API, can we make this:

```
using QuantizationTable = Array<u16, 64>;
using QuantizationTables = Array<QuantizationTable, 4>;
```

The Optional<> adds very little, and in public API it's even more awkward.

Reply:
> The Optional<> adds very little

How can we replace it then? I don't see a solution that is objectively better.

Reply:
Replace in what sense? If "with what type": see my snippet. If "with what value": assigning to an optional also copies the entire array.

(Sorry, I'm almost certain I'm misunderstanding the question!)

Reply:
I didn't create an API with a funky signature for fun, this is the internal type that is used in the decoder.
Your solution doesn't conserve all the features of the old type, so it can't be taken as-is.

And I don't see a solution that will both provide both a nice API and a clear code in the decoder.

Reply:
The only use is to emit a diag if the table is missing in the file, right? What does libjpeg do for that? The spec says "behavior is not specified" or something like that iirc, so just initializing with 0 would be fine per spec.